demo-circuit/src/naive_threshold_sig_w_key_rotation/constraints/data_structures.rs

```diff
@@ -121,7 +121,7 @@ impl ValidatorKeysUpdatesGadget {
                 cs.ns(|| format!("alloc_fe gadget elems for mkey_{}", i)),
             )?;
             let master_key_fe_hash_g = FieldHashGadget::enforce_hash_constant_length(
-                cs.ns(|| format!("H(mkey_{})", i)),
+                cs.ns(|| format!("H(mkey_{}||'m'||CONST_SALT||epoch_id||ledger_id)", i)),
```

Comment:
It should be H(H(mkey_{}), ...)

Reply:
Thanks for pointing it out, I changed it when computing the message to sign but not here. I changed it to `H(mkey)`, because we are computing the leaf of the Merkle-tree where we don't concatenate further data to the master key